packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -993,6 +993,99 @@ class DefaultTextStyleTransition extends AnimatedWidget {
 /// ** See code in examples/api/lib/widgets/transitions/animated_builder.0.dart **
 /// {@end-tool}
 ///
+/// {@template flutter.flutter.animatedbuilder_changenotifier.rebuild}
+/// ## Granular rebuilds with AnimatedBuilder and ChangeNotifier
+///
+/// A [State.setState] method call requires Flutter to entirely rebuild the
+/// current widget plus a variable portion of the subtree, depending on the
+/// depth and the structure.
+///
+/// To avoid unnecessary work, the [AnimatedBuilder] class can be used along
+/// with a [ChangeNotifier] to only rebuild certain portions a widget,
+/// leaving other parts and the subtree untouched.
+///
+/// Since a [ChangeNotifier] is a subtype of [Listenable], it can be used in
+/// an [AnimatedBuilder] to listen for changes. For example, to keep track
+/// of the state of a simple counter, a [ChangeNotifier] can be created and
+/// exposed to the widgets below using an [InheritedWidget].
+///
+/// {@tool snippet}
+///
+/// This is an example that shows how an [InheritedWidget] can expose a
+/// [ChangeNotifier] to a widget and its children.
+///
+/// ```dart
+/// class Counter extends ChangeNotifier {
+///   int _counter;
+///
+///   Counter({int value = 0}) : _counter = value;
+///
+///   int get value => _counter;
+///   set value(int value) {
+///     if (_counter != value) {
+///       _counter = value;
+///       notifyListeners();
+///     }
+///   }
+/// }
+///
+/// class CounterWidget extends InheritedWidget {
+///   const CounterWidget({
+///     Key? key,
+///     required this.counter,
+///     required Widget child,
+///   }) : super(key: key, child: child);
+///
+///   final Counter counter;
+///
+///   static CounterWidget of(BuildContext context) {
+///     final result = context.dependOnInheritedWidgetOfExactType<CounterWidget>();
+///     assert(result != null, 'No CounterWidget found in context');
+///     return result!;
+///   }
+///
+///   @override
+///   bool updateShouldNotify(CounterWidget oldWidget) {
+///     return counter.value != oldWidget.counter.value;
+///   }
+/// }
+/// ```
+/// {@end-tool}
+///
+/// In the following example, `CounterPage` is a child of `CounterWidget` so
```

Comment:
The examples should probably just be merged with a focus on the AnimationBuilder.